README.md

```diff
@@ -6,9 +6,29 @@
 * ✔ Covers both fixed-task and variable-task scenarios
 * ✔ Supports flexible pool sizes
 * ✔ Proper async tracking integration
+* ✔ Tracking statistics for run and wait times
+* ✔ Cancelation Support
 
 [MIT Licensed][].
 
+## Introduction
+
+Piscina works by creating a pool of Node.js Worker Threads to which
+one or more tasks may be dispatched. Each worker thread executes a
+single exported function defined in a separate file. Whenever a
+task is dispatched to a worker, the worker invokes the exported
+function and reports the return value back to Piscina when the
+function completes.
+
+## Current Limitations (Things we're working on / would love help with)
```

Comment:
Can we avoid putting this before the example? I feel somewhat strongly about an introductory example being one of the very first things in a README :)

Reply:
Missed these :-) ... will make these edits now